pkg/core/kustomize/kustomize.go

```diff
@@ -0,0 +1,141 @@
+package kustomize
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sigs.k8s.io/kustomize/k8sdeps"
+	"sigs.k8s.io/kustomize/pkg/commands/build"
+	"sigs.k8s.io/kustomize/pkg/fs"
+	"time"
+)
+
+type Kustomizer interface {
+	// Applies the provided label to the provided remote or local resource definition
+	// Returns the customized resource contents
+	// Returns an error if
+	// - the URL scheme is not supported (only file, http and https are)
+	// - retrieving the content fails
+	// - applying the customization fails
+	ApplyLabel(resourceUri *url.URL, label *Label) ([]byte, error)
+}
+
+type Label struct {
+	Name  string
+	Value string
+}
+
+func (l *Label) AsMap() map[string]string {
+	return map[string]string{l.Name: l.Value}
+}
+
+type kustomizer struct {
+	fakeDir    string
+	fs         fs.FileSystem
+	httpClient *http.Client
+}
+
+func MakeKustomizer(timeout time.Duration) Kustomizer {
+	return &kustomizer{
+		fs:      fs.MakeFakeFS(),
```

Comment:
Probably worth a comment about why a _fake_ FS is being used. Is it really just an in-memory FS?

Reply:
No strong reasons for it, I'm open to revisit that assumption.
I started with that as the [other integration](https://github.com/kubernetes-sigs/kind/pull/77) did the same.
I also find it easier to reason about, as we do not need to deal with permissions and other OS peculiarities. Moreover, it seems we do not process huge documents at the moment.